### ISSUE TITLE
Bug when trying to update product with variants through API Rest

### ISSUE THREAD

**Issue by c3do** [Dolibarr/dolibarr#12397]
The result is null when we try to update a product with variations through API Rest

How to reproduce:
- Create a product
- Add a variation
- Update product through API Rest

Log:
```
2019-11-09 12:28:35 ERR     127.0.0.1          DoliDBMysqli::query SQL Error query: INSERT INTO llx_product_price(price_level,date_price, fk_product, fk_user_author, price, price_ttc, price_base_type,tosell, tva_tx, default_vat_code, recuperableonly, localtax1_tx, localtax2_tx, localtax1_type, localtax2_type, price_min,price_min_ttc,price_by_qty,entity,fk_price_expression)  VALUES(1, '2019-11-09 12:28:35',6,0,29,34.8,'',1,20, null,0, 0, 0, '0', '0', 0,0,0,1,null)
2019-11-09 12:28:35 ERR     127.0.0.1          DoliDBMysqli::query SQL Error message: DB_ERROR_1452 Cannot add or update a child row: a foreign key constraint fails (`dolibarr11`.`llx_product_price`, CONSTRAINT `fk_product_price_product` FOREIGN KEY (`fk_user_author`) REFERENCES `llx_user` (`rowid`))
2019-11-09 12:28:35 ERR     127.0.0.1          Error url=/dolibarr11/htdocs/api/index.php/products/2, query_string=, sql=INSERT INTO llx_product_price(price_level,date_price, fk_product, fk_user_author, price, price_ttc, price_base_type,tosell, tva_tx, default_vat_code, recuperableonly, localtax1_tx, localtax2_tx, localtax1_type, localtax2_type, price_min,price_min_ttc,price_by_qty,entity,fk_price_expression)  VALUES(1, '2019-11-09 12:28:35',6,0,29,34.8,'',1,20, null,0, 0, 0, '0', '0', 0,0,0,1,null), db_error=Cannot add or update a child row: a foreign key constraint fails (`dolibarr11`.`llx_product_price`, CONSTRAINT `fk_product_price_product` FOREIGN KEY (`fk_user_author`) REFERENCES `llx_user` (`rowid`))
```

**Comment by dpriskorn** [comment]
Could you test if its fixed in v12?

**Comment by c3do** [comment]
Normally i fixed it with the patch #12398

**Comment by dpriskorn** [comment]
can you please close this if its fixed?
